docs/adding-new-cloud-provider.md

```diff
@@ -0,0 +1,50 @@
+# How to add a new Cloud Provider
+
+## Add provider specific code to the cloud credentials operator 
+The Cloud Credentials Operator has three modes
+
+- *Manual* is a recommended mode. In this mode a user manually performs the job of the cloud credential operator.  
+- *Mint* mode. In this mode the admin credential is stored in kube-system namespace and then used to create fine-grained permission for in-cluster components.
+- *Passthrough* mode. In this mode a single credential having all the required permissions is stored in kube-system namespace and passed to other in-cluster components. This credential cannot mint additional credentials.  
+
+We need a new provider to support at least the *Manual* mode. The other two modes are optional. Learn more about CCO modes [here](../README.md#modes-of-operation). 
+
+For adding a new provider we need new code at the following places
+
+1) Create a ProviderSpec API that holds the specifications for the credentials request. Refer aws implementation [here](../pkg/apis/cloudcredential/v1/types_aws.go).
+2) Create a ProviderStatus API that holds the status of the credentials request. Refer aws implementation [here](../pkg/apis/cloudcredential/v1/types_aws.go).
+3) Register the above two in [register.go](../pkg/apis/cloudcredential/v1/register.go).
+
+Follow the next steps only if you are planning to support *Mint* and *Passthrough* mode.  
+
+4) Implement [this](../pkg/operator/credentialsrequest/actuator/actuator.go) actuator interface. Here you add code to create/update/manage credentials on the cloud. Refer aws implementation [here](../pkg/aws/actuator/actuator.go).
+5) Add the new actuator to the [controller.go](../pkg/operator/controller.go)
+6) Add the new platform in `crInfraMatches` function in [credentrialsrequest_controller.go](../pkg/operator/credentialsrequest/credentialsrequest_controller.go) so that it identified as a valid platform by the controller.
+7) Add the code for [secretannotator_controller.go](../pkg/operator/secretannotator/secretannotator_controller.go) to annotate the cloud credential secret to indicate the credential's capabilities. Refer aws implementation [here](../pkg/operator/secretannotator/aws/reconciler.go).
```

Comment:
Optionally? I think this is optional b/c we have several platforms that only support Passthrough mode, and they don't have a secret_annotator implementation.

Reply:
We do have some platforms (like OpenStack) that only support Passthrough mode and also have secret annotator implementation. Should we just avoid this controller for Passthough only cloud platform? Whats the harm in keeping it just for consistency?

Reply:
Let's go ahead an leave it in as we should have the annotator behave as similar as possible across platforms.